test/libevmasm/Optimiser.cpp

```diff
@@ -967,6 +967,29 @@ BOOST_AUTO_TEST_CASE(peephole_swap_comparison)
 	}
 }
 
+BOOST_AUTO_TEST_CASE(peephole_truthy_and)
+{
+  AssemblyItems items{
+    AssemblyItem(PushTag, 1),
```

Comment:
You have to use Tag and PushTag the other way round. `Tag` is the tag / jumpdest itself and `PushTag` is a reference to the tag.

Reply:
Ah that makes sense. So is this what in infinite loop looks like?

```
AssemblyItem(Tag, 1)
AssemblyItem(PushTag, 1)
Instruction::JUMP
```

Reply:
Yes!